sarracenia/config.py

```diff
@@ -1750,10 +1747,21 @@ def _resolveQueueName(self,component,cfg):
             if self.action not in [ 'start', 'foreground', 'declare' ]:
                 return
 
-            if (self.queueName is not None) and (not hasattr(self,'no') or (self.no < 2)):
-                f = open(queuefile, 'w')
-                f.write(self.queueName)
-                f.close()
+            # first make sure directory exists.
+            if not os.path.isdir(os.path.dirname(queuefile)):
+                pathlib.Path(os.path.dirname(queuefile)).mkdir(parents=True, exist_ok=True)
+
+            if not os.path.isfile(queuefile) and (self.queueName is not None): 
```

Comment:
Is this what we want? Only write when the `queuefile` doesn't exist?
What if the `queuefile` is corrupt? It wouldn't get rewritten

Reply:
if the queuefile is corrupt... something weird is going on... you get rid of it with 'cleanup'.   or you can stop the flow and manually delete just the file, but if the relationship with the queue is gone... state is kind of pooched... so cleanup makes sense.